### PR TITLE
DEVEXP-555: Rebasing k8s.io

### PR DESCRIPTION
**Pull request by ricardomaraschini**

### CODE REVIEW

**Comment by ricardomaraschini** [comment]
/retest

Error says:
```
(usually a result of a race or resource pressure. re-running the job should help)
```

**Comment by ricardomaraschini** [comment]
Tests are failing, waiting for https://github.com/openshift/release/pull/8109 to get merged.

**Comment by coreydaley** [comment]
Waiting on https://github.com/openshift/release/pull/8224

**Comment by coreydaley** [comment]
I don't think we have a standard for line length in this repository, or openshift as a whole, but I feel like there are a lot of unnecessary line breaks in this pull request.

Maybe only break them up if there are more than three or they are really long?

**Comment by ricardomaraschini** [comment]
> I don't think we have a standard for line length in this repository, or openshift as a whole, but I feel like there are a lot of unnecessary line breaks in this pull request.
> 
> Maybe only break them up if there are more than three or they are really long?

All new break lines removed. PTAL.

File: pkg/dockerregistry/server/auth.go [hunk 1]
```diff
@@ -375,7 +375,7 @@ func getOpenShiftAPIToken(req *http.Request) (string, error) {
 }
 
 func verifyOpenShiftUser(ctx context.Context, c client.UsersInterfacer) (string, string, error) {
-	userInfo, err := c.Users().Get("~", metav1.GetOptions{})
+	userInfo, err := c.Users().Get(context.TODO(), "~", metav1.GetOptions{})
```
Comment (dmage) [inline pkg/dockerregistry/server/auth.go:378]: Use `ctx` instead of `context.TODO()`.

File: pkg/dockerregistry/server/signaturedispatcher.go [hunk 2]
```diff
@@ -120,7 +120,9 @@ func (s *signatureHandler) Put(w http.ResponseWriter, r *http.Request) {
 	newSig := &imageapiv1.ImageSignature{Content: sig.Content, Type: sig.Type}
 	newSig.Name = sig.Name
 
-	_, err = client.ImageSignatures().Create(newSig)
+	_, err = client.ImageSignatures().Create(
```
Comment (dmage) [inline pkg/dockerregistry/server/signaturedispatcher.go:123]:
```suggestion
	_, err = client.ImageSignatures().Create(s.ctx, newSig, metav1.CreateOptions{})
```

File: pkg/dockerregistry/server/auth.go (hunk 1 above)
Comment (ricardomaraschini) [reply pkg/dockerregistry/server/auth.go:378]: Good catch!

Comment (ricardomaraschini) [reply pkg/dockerregistry/server/auth.go:378]: I think that this problem happens in a lot of different places, I will fix them and submit a new commit. Please hold on.

File: pkg/dockerregistry/server/signaturedispatcher.go (hunk 2 above)
Comment (ricardomaraschini) [reply pkg/dockerregistry/server/signaturedispatcher.go]: done.

**Comment by adambkaplan** [comment]
> I don't think we have a standard for line length in this repository, or openshift as a whole, but I feel like there are a lot of unnecessary line breaks in this pull request.

Sadly @coreydaley I don't think `gofmt` enforces a line width, and there be dragons if we adopt something that strays too far from the golang ecosystem. I think the rule of thumb is start adding line breaks at 100 characters to make it readable on GitHub.

**Comment by dmage** [comment]
@adambkaplan @coreydaley @ricardomaraschini we follow Go Code Review Comments, it has a note about line length: https://github.com/golang/go/wiki/CodeReviewComments#line-length

Long lines is a code small and wrapping just deodorizes them. So I'd agree with @coreydaley and prefer not to wrap lines in the middle of function calls, especially lines that are less than 100-120 characters (thanks @ricardomaraschini for fixing it).

File: pkg/dockerregistry/server/auth.go (hunk 1 above)
Comment (ricardomaraschini) [reply pkg/dockerregistry/server/auth.go]: I sent a new commit that address context's usage, ptal.

**Pull request merged**